Mention consequences of specify PublishTrimmed (#30450)
dotnet/docs: docs/core/deploying/trimming/trimming-options.md

  Enable trimming during publish. This also turns off trim-incompatible features and enables [trim analysis](#roslyn-analyzer) during build.

> **Note:** If you specify that property from command line, your debugging experience would differ and you may encounter additional bugs in final product.

Reviewer: ```suggestion
> [!NOTE]
> If you specify trimming as enabled from the command line, your debugging experience will differ and you may encounter additional bugs in the final product.

```
